src/pages/index.js

```diff
@@ -0,0 +1,327 @@
+      })
+      .catch((err) => {
+        console.error('Error updating profile:', err);
+      })
+      .finally(() => {
+        saveButton.textContent = 'Save';
+        saveButton.disabled = false;
+      });
+  }
+
+
+  function handleNewPostSubmit(evt) {
+    evt.preventDefault();
+    const name = newPostCaptionInput.value;
+    const link = newPostLinkInput.value;
+
+    api.createCard(name, link)
+      .then((newCardData) => {
+        const cardElement = getCardElement(newCardData);
+        cardsContainer.prepend(cardElement);
+        closeModal(newPostModal);
+        newPostForm.reset();
+        disableButton(cardSubmitBtn, settings);
+      })
+      .catch((err) => {
+        console.error('Error creating new card:', err);
+      });
+  }
+
+  function openModal(modal) {
+    modal.classList.add('modal_opened');
+
+    const closeModalOverlayClick = (event) => {
+      if (event.target === modal) {
+        closeModal(modal);
+      }
+    };
+    modal.addEventListener('click', closeModalOverlayClick);
+
+    const closeModalOnEsc = (evt) => {
+      if (evt.key === 'Escape') {
+        closeModal(modal);
+      }
+    };
+    document.addEventListener('keydown', closeModalOnEsc);
+
+    modal._closeModalOnClickOutside = closeModalOverlayClick;
+    modal._closeModalOnEsc = closeModalOnEsc;
+  }
+
+  function closeModal(modal) {
+    modal.classList.remove('modal_opened');
+
+    modal.removeEventListener('click', modal._closeModalOnClickOutside);
+    document.removeEventListener('keydown', modal._closeModalOnEsc);
+
+    delete modal._closeModalOnClickOutside;
+    delete modal._closeModalOnEsc;
+  }
+
+  editButton.addEventListener('click', () => {
+    resetValidation(editProfileForm, [editModalNameInput, editModalDescriptionInput]);
+    openModal(editProfileModal);
+    editModalNameInput.value = profileName.textContent;
+    editModalDescriptionInput.value = profileDescription.textContent;
+  });
+
+  newPostButton.addEventListener('click', () => {
+    openModal(newPostModal);
+  });
+
+  newPostCloseButton.addEventListener('click', () => {
+    closeModal(newPostModal);
+  });
+
+  editAvatarCloseButton.addEventListener('click', () => {
+    closeModal(editAvatarModal);
+  });
+
+  editProfileCloseButton.addEventListener('click', () => {
+    closeModal(editProfileModal);
+  });
+
+  previewCloseButton.addEventListener('click', () => {
+    closeModal(previewModal);
+  });
+
+  deleteModalCloseButton.addEventListener('click', () => {
+    closeModal(deleteModal);
+  });
+
+  deleteModalCancelButton.addEventListener('click', () => {
+    closeModal(deleteModal);
+  });
+
+  document.querySelector('.profile__avatar-edit-btn').addEventListener('click', openAvatarModal);
+  editAvatarForm.addEventListener('submit', handleEditAvatarSubmit);
+  deleteModalConfirmButton.addEventListener('click', handleDeleteCard);
+  editProfileCloseButton.addEventListener('click', () => { closeModal(editProfileModal) });
```

Comment:
`  editProfileCloseButton.addEventListener('click', () => { closeModal(editProfileModal) });
`

It's not needed because you handle the profile close icon in line 298